staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -44,6 +44,9 @@ const (
 	// resourceVersionTooHighRetrySeconds is the seconds before a operation should be retried by the client
 	// after receiving a 'too high resource version' error.
 	resourceVersionTooHighRetrySeconds = 1
+
+	// eventRecordDuration is time duration of events we want to keep.
+	eventFreshDuration = 5 * time.Minute
```

Comment:
As mentioned by @lavalamp in the bug, we need a maximum cache size that we can't exceed.

We probably also want a minimum.

As for the values, we need some kind of heuristic.
My current best estimation is assume that leases are the most frequent and happen once per 10s.
For 5k-node clusters, that gives 5000*6=30k per minute, so 150k per 5 minutes. So set the max to 150k.

Reply:
Can we set lower bound as 100(minimum in [NewHeuristicWatchCacheSizes](https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/cachesize/cachesize.go#L25))

Reply:
I would like to go lower than 100, but for now maybe 100 is good enough, as we won't regress anyway.

Reply:
I'd like to change default lower or upper bound if there is a better value.